graylog2-web-interface/src/components/errors/ReportedError.test.jsx

```diff
@@ -0,0 +1,95 @@
+// @flow strict
+import React from 'react';
+import { render, cleanup, waitForElement, waitForElementToBeRemoved } from 'wrappedTestingLibrary';
+
+import suppressConsole from 'helpers/suppressConsole';
+import ErrorsActions from 'actions/errors/ErrorsActions';
+import { createReactError, createUnauthorizedError } from 'logic/errors/ReportedErrors';
+import { FetchError } from 'logic/rest/FetchProvider';
+import ReportedError from './ReportedError';
+
+jest.unmock('logic/rest/FetchProvider');
+jest.mock('react-router', () => ({ withRouter: (x) => x }));
+
+const router = {
+  listen: () => jest.fn(),
+};
+
+describe('ReportedError', () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  it('registers to router upon mount', () => {
+    const mockRouter = {
+      listen: jest.fn(() => jest.fn()),
+    };
+    render(<ReportedError router={mockRouter}>Hello World!</ReportedError>);
+
+    expect(mockRouter.listen).toHaveBeenCalledTimes(1);
+  });
+
+  it('unregisters from router upon unmount', () => {
+    const unlisten = jest.fn();
+    const mockRouter = {
+      listen: () => unlisten,
+    };
+    const { unmount } = render(<ReportedError router={mockRouter}>Hello World!</ReportedError>);
+
+    unmount();
+
+    expect(unlisten).toHaveBeenCalled();
+  });
+
+
+  it('displays child component if there is no error', () => {
+    const { getByText } = render(<ReportedError router={router}>Hello World!</ReportedError>);
+
+    expect(getByText('Hello World!')).not.toBeNull();
+  });
+
+  it('displays runtime error page when react error got reported', () => {
+    suppressConsole(async () => {
+      const { getByText } = render(<ReportedError router={router}>Hello World!</ReportedError>);
+
+      ErrorsActions.report(createReactError(new Error('The error message'), { componentStack: 'The component stack' }));
+
+      await waitForElementToBeRemoved(() => getByText('Hello World!'));
+      await waitForElement(() => getByText('Something went wrong.'));
```

Comment:
`waitForElement` is [deprecated](https://testing-library.com/docs/dom-testing-library/api-async#waitforelement-deprecated-use-find-queries-or-waitfor) in the next version of the react testing-library and `waitFor` is a plugin replacement, so I think we should already start using it in new code.